mcs/tools/resgen/monoresgen.cs

```diff
@@ -371,17 +406,86 @@ class TxtResourceReader : IResourceReader {
 	public IDictionaryEnumerator GetEnumerator() {
 		return data.GetEnumerator ();
 	}
-	
+
+	public static string NextWord(ref string line) {
+		int i, j;
+		string keywd;
+		line = line.TrimStart ();
+		for (i = 0; i < line.Length && !Char.IsWhiteSpace (line [i]) && line [i] != ';'; )
+			i++;
```

Comment:
Like this?

     for(i = 0; ....; i++);

Do you have any standard for empty `for` loops? Easy to miss semicolon at the end...